airbyte/cloud/_workspaces.py

```diff
@@ -0,0 +1,351 @@
+            if not cache._deployed_destination_id:  # noqa: SLF001
+                raise ValueError("Cache has not been deployed.")  # noqa: TRY003
+
+            destination = cache._deployed_destination_id  # noqa: SLF001
+
+        if destination is None:
+            raise ValueError("No destination ID provided.")  # noqa: TRY003
+
+        delete_destination(
+            destination_id=destination,
+            api_root=self.api_root,
+            api_key=self.api_key,
+        )
+
+    # Deploy and delete connections
+
+    def deploy_connection(
+        self,
+        source: Source | str,
+        cache: CacheBase | None = None,
+        destination: str | None = None,
+        table_prefix: str | None = None,
+        selected_streams: list[str] | None = None,
+    ) -> str:
+        """Deploy a source and cache to the workspace as a new connection.
+
+        Returns the newly deployed connection ID as a `str`.
+
+        Args:
+            source (Source | str): The source to deploy. You can pass either an already deployed
+                source ID `str` or a PyAirbyte `Source` object. If you pass a `Source` object,
+                it will be deployed automatically.
+            cache (CacheBase, optional): The cache to deploy as a new destination. You can provide
+                `cache` or `destination`, but not both.
+            destination (str, optional): The destination ID to use. You can provide
+                `cache` or `destination`, but not both.
+        """
+        # Resolve source ID
+        source_id: str
+        if isinstance(source, Source):
+            selected_streams = selected_streams or source.get_selected_streams()
+            if source._deployed_source_id:  # noqa: SLF001
+                source_id = source._deployed_source_id  # noqa: SLF001
+            else:
+                source_id = self.deploy_source(source)
+        else:
+            source_id = source
+            if not selected_streams:
+                raise exc.PyAirbyteInputError(
+                    guidance="You must provide `selected_streams` when deploying a source ID."
+                )
+
+        # Resolve destination ID
+        destination_id: str
+        if destination:
+            destination_id = destination
+        elif cache:
+            table_prefix = table_prefix if table_prefix is not None else (cache.table_prefix or "")
+            if not cache._deployed_destination_id:  # noqa: SLF001
+                destination_id = self.deploy_cache_as_destination(cache)
+            else:
+                destination_id = cache._deployed_destination_id  # noqa: SLF001
+        else:
+            raise exc.PyAirbyteInputError(
+                guidance="You must provide either a destination ID or a cache object."
+            )
+
+        assert source_id is not None
+        assert destination_id is not None
+
+        deployed_connection = create_connection(
+            name="Connection (Deployed by PyAirbyte)",
```

Comment:
Currently the connection names (in cloud UI) look like "Github -> Google Sheets". should we try to stick to the same format, so something like "Github -> Google Sheets (Deployed by PyAirbyte)"

Reply:
Good idea! At this point in the code though, I only have the GUID IDs. I do think it's worth a lookup operation though, so the names are readable - and the calls themselves are quick.